pkg/crcontrollers/lvg/lvgcontroller.go

```diff
@@ -324,3 +332,20 @@ func (c *Controller) increaseACSize(driveID string, size int64) {
 
 	ll.Errorf("Corresponding AC for drive ID %s not found", driveID)
 }
+
+// resetACSize sets size of corresponding AC to 0 to avoid further allocations
+func (c *Controller) resetACSizeOfLVG(lvgName string) {
+	// read AC
+	if ac := c.crHelper.GetACByLocation(lvgName); ac != nil {
+		// update if not null already
+		if ac.Spec.Size != 0 {
+			ac.Spec.Size = 0
+			if err := c.k8sClient.UpdateCR(context.Background(), ac); err != nil {
+				c.log.Errorf("Unable to set AC CR %s size to 0, error: %v.", ac.Name, err)
```

Comment:
I think it will be better to handle this error correctly. We can safely return error and let CR controller to do a retry.
Also, it will be nice if we can modify crHelper.GetACByLocation method to be able to distinguish "not found" and "request error". If AC read failed we can return error and allow reconciliation to restart.

Reply:
Ok, thanks. Will refactor this place

Reply:
Refactored. Please take a look.